pkg/kube/diff.go

```diff
@@ -107,6 +107,67 @@ func renderObj(obj runtime.Object, gvk *schema.GroupVersionKind, renderYaml bool
 	return string(yamlBytes), nil
 }
 
+// removeSpuriousDiff implements conditional field removal from live and/or head
+// object depending on the value of the field and the difference between live
+// and head. This behavior differs from that of using the
+// diffFilters, which is unconditional removal.
+//
+// Specifically, this function removes:
+// * Master-assigned NodePort on Live object if head does not specify a NodePort.
+// * Kubernetes namespace finalizer on live and head objects.
+func removeSpuriousDiff(live, head runtime.Object) (newLive, newHead runtime.Object) {
+	newLive, newHead = live.DeepCopyObject(), head.DeepCopyObject()
+	removeSpuriousNodePortDiff(&newLive, &newHead)
+	removeSpuriousNamespaceFinalizerDiff(&newLive, &newHead)
+	return
+}
+
+func removeSpuriousNodePortDiff(live, head *runtime.Object) {
```

Comment:
it is better not to use pointers for the interfaces. Otherwise, we'll need to add some nil checks here when one of these objects is nil. Some dummy code related with this: https://play.golang.org/p/waGCj--3B0X

Reply:
Thanks for the link. I have just added checks for nil pointers. I am using pointers to interface here so that I can just clone once and make the following edits in place ( i.e. due to performance concern).

Reply:
Actually, having a pointer to an interface doesn't seem to be necessary. You can take a look at [this](https://stackoverflow.com/a/44372954) SO response. I've also changed the initial example a bit [here](https://play.golang.org/p/XEyP87kE7dI) to support the functionality behind non-pointer interfaces.

Reply:
Thanks for the link! It is super helpful